docs/reference/snapshot-restore/apis/restore-snapshot-api.asciidoc

```diff
@@ -0,0 +1,230 @@
+supports referencing the original text, according to the https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html#appendReplacement-java.lang.StringBuffer-java.lang.String-[`appendReplacement`] logic.
++
+The request will fail if two or more data streams or indices will be renamed into the same name.
++
+include::{es-ref-dir}/snapshot-restore/restore-snapshot.asciidoc[tag=rename-restored-data-stream-tag]
+
+[[restore-snapshot-api-rename-replacement]]
+`rename_replacement`::
+(Optional, string)
+Defines the rename replacement string. See <<restore-snapshot-api-rename-pattern,`rename_pattern`>> for more information.
+
+`wait_for_completion`::
+(Optional, boolean)
+If `true`, the request returns a response when the restore operation
+completes.
++
+If `false`, the request returns a response when the restore operation initializes.
+Defaults to `false`.
+
+[[restore-snapshot-api-example]]
+==== {api-examples-title}
+
+The following request restores `index_1` and `index_2` from `snapshot_2`. The `rename_pattern` parameter indicates that any index matching the regex pattern `index_(.+)` will be renamed using the pattern `restored_index_$1`, which is defined by the `rename_replacement` parameter.
+
+For example, `index_1` will be renamed to `restored_index_1` and `index_2` will be renamed to `restored_index_2`.
+
+[source,console]
+----
+POST /_snapshot/my_repository/snapshot_2/_restore?wait_for_completion=true
+{
+  "indices": "index_1,index_2",
+  "ignore_unavailable": true,
+  "include_global_state": false,
+  "rename_pattern": "index_(.+)",
+  "rename_replacement": "restored_index_$1",
+  "include_aliases": false
+}
+----
+
+The API returns an acknowledgement if the request succeeds. If the request encounters errors, the response indicates any issues found, such as
+open indices that are blocking the restore operation from completing.
+
+For example, the following error indicates that `snapshot_2` cannot be restored because an open index with the same name (`restored_index_2`) already exists in the cluster.
+
+[source,console-result]
+----
+{
+  "error" : {
+    "root_cause" : [
+      {
+        "type" : "snapshot_restore_exception",
+        "reason" : "[my_repository:snapshot_2/0LIMUElxQkyRoimBDAkYIQ]
+        cannot restore index [restored_index_2] because an open index
+        with same name already exists in the cluster. Either close or
+        delete the existing index or restore the index under a different
+        name by providing a rename pattern and replacement name"
+      }
+    ],
+    "type" : "snapshot_restore_exception",
+    "reason" : "[my_respository:snapshot_2/0LIMUElxQkyRoimBDAkYIQ]
+    cannot restore index [restored_index_2] because an open index
+    with same name already exists in the cluster. Either close or
+    delete the existing index or restore the index under a different
+    name by providing a rename pattern and replacement name"
+  },
+  "status" : 500
+}
+----
+// TEST[skip:example only]
```

Comment:
I'd drop this portion of the example.

With a few exceptions, we don't document error messages. Two reasons:

1. Errors aren't typically an outcome any user wants. It's usually more helpful to document ways to prevent errors in the first place.

2. If the error message changes, we have no way of catching that in these docs. Because you skipped the response here, no CI failure would be raised. That's not the case for most requests/responses.

For example, if a response added a new property, engineers would need to update any existing doc snippets containing that response to include the new property. Otherwise the CI would fail on their PR.
Skipping tests generally undermines that system.

```suggestion
```

Reply:
Thanks for the context James. I was uncertain about adding the error message and whether it would create more ambiguity. My thoughts were to provide some insight into a common error that users might see, but I agree with your assessment regarding broken CI builds. I'll remove the highlighted section.